File: backend/routes/ai.py
# ================================================================
# SmartPayroll AI — AI Routes (all AI features via backend)
# ================================================================

from fastapi import APIRouter, HTTPException
from models import (
    AIChatRequest, AISalaryRequest, AIAttendanceRequest,
    AIPayrollReviewRequest, AIReportRequest,
)
from services import ai_service

router = APIRouter(prefix="/api/ai", tags=["AI"])


@router.post("/chat")
async def ai_chat(data: AIChatRequest):
    """Chat with AI payroll assistant."""
    try:
        response = await ai_service.chat(data.message, data.history)
        return {"response": response}
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))


@router.post("/insights")
async def ai_insights():
    """Generate dashboard insights."""
    try:
        insights = await ai_service.generate_insights()
        return {"response": insights}
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))


@router.post("/salary-suggest")
async def ai_salary_suggest(data: AISalaryRequest):
    """AI salary structure suggestion."""
    try:
        suggestion = await ai_service.suggest_salary(data.role, data.department, data.budget)
        return {"response": suggestion}
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))


@router.post("/attendance-analyze")
async def ai_attendance_analyze(data: AIAttendanceRequest):
    """AI attendance anomaly detection."""
    try:
        analysis = await ai_service.analyze_attendance(data.employeeId, data.month)
        return {"response": analysis}
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))


@router.post("/payroll-review")
async def ai_payroll_review(data: AIPayrollReviewRequest):
    """AI payroll review."""
    try:
        review = await ai_service.review_payroll(data.payrollId)
        return {"response": review}
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))


@router.post("/payslip-summary/{payroll_id}")
async def ai_payslip_summary(payroll_id: str):
    """Generate payslip AI summary."""
    try:
        summary = await ai_service.generate_payslip_summary(payroll_id)
        return {"response": summary}
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))


@router.post("/report-summary")
async def ai_report_summary(data: AIReportRequest):
    """Generate report executive summary."""
    try:
        summary = await ai_service.generate_report_summary(data.reportType, data.data)
        return {"response": summary}
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))


@router.post("/bonus-recommend")
async def ai_bonus_recommend():
    """AI bonus recommendation."""
    try:
        recommendation = await ai_service.recommend_bonus()
        return {"response": recommendation}
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))

File: src/ai.js
// ================================================================
// SmartPayroll AI — AI Engine (Backend API Proxy)
// ================================================================

import store from './store.js';

const API_BASE = import.meta.env.VITE_API_URL || 'http://localhost:8000';
const RUNTIME_HOST = typeof window !== 'undefined' ? window.location.hostname : 'localhost';
const IS_HOSTED_RUNTIME = !['localhost', '127.0.0.1'].includes(RUNTIME_HOST);

function isLocalApiConfiguredOnHostedRuntime() {
  try {
    const apiHost = new URL(API_BASE).hostname;
    return IS_HOSTED_RUNTIME && ['localhost', '127.0.0.1'].includes(apiHost);
  } catch {
    return false;
  }
}

async function aiApi(path, body = {}) {
  if (isLocalApiConfiguredOnHostedRuntime()) {
    throw new Error('AI backend is configured as localhost, which is unreachable from GitHub Pages. Set VITE_API_URL to a public backend URL.');
  }

  let response;
  try {
    response = await fetch(`${API_BASE}/api/ai${path}`, {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify(body),
    });
  } catch (error) {
    if (error.message?.includes('Failed to fetch') || error.message?.includes('NetworkError')) {
      throw new Error('Unable to reach backend API. Verify VITE_API_URL points to a live public backend.');
    }
    throw error;
  }

  if (!response.ok) {
    const error = await response.text();
    let detail = error;
    try { detail = JSON.parse(error).detail || error; } catch {}
    throw new Error(detail);
  }

  const data = await response.json();
  return data.response || data;
}

class AIEngine {
  /**
   * Chat with the AI payroll assistant
   */
  async chat(userMessage) {
    const history = store.getChatHistory().slice(-10);
    return await aiApi('/chat', { message: userMessage, history });
  }

  /**
   * Generate dashboard insights
   */
  async generateDashboardInsights() {
    return await aiApi('/insights');
  }

  /**
   * Suggest salary structure for a role
   */
  async suggestSalaryStructure(role, department, budget) {
    return await aiApi('/salary-suggest', { role, department, budget });
  }

  /**
   * Detect attendance anomalies
   */
  async detectAttendanceAnomalies(employeeId, month) {
    return await aiApi('/attendance-analyze', { employeeId, month });
  }

  /**
   * Review payroll for anomalies
   */
  async reviewPayroll(payrollData) {
    return await aiApi('/payroll-review', { payrollId: payrollData.id });
  }

  /**
   * Generate payslip summary
   */
  async generatePayslipSummary(payrollData) {
    return await aiApi(`/payslip-summary/${payrollData.id}`);
  }

  /**
   * Generate report executive summary
   */
  async generateReportSummary(reportType, data) {
    return await aiApi('/report-summary', { reportType, data });
  }

  /**
   * Smart bonus recommendation
   */
  async recommendBonus() {
    return await aiApi('/bonus-recommend');
  }
}

export const ai = new AIEngine();
export default ai;
